### PR TITLE
support removing specific attached events

### PR DESCRIPTION
**Pull request by jsdw**
This is a feature I needed to make ResizeSensor.js (as a standalone lib) useful for me at work; its only a small addition and seems like others would find it useful too :)

Basically, this allows you to pass the original event in to `.detach` methods as well as the element (fully optional) to only remove that event. from the queue.

### CODE REVIEW

File: src/ResizeSensor.js [hunk 1]
```diff
@@ -163,19 +175,23 @@
...
             }
         };
     };
 
-    ResizeSensor.detach = function(element) {
+    ResizeSensor.detach = function(element, ev) {
+        if(typeof ev == "function"){
+            element.resizedAttached.remove(ev);
+            if(element.resizedAttached.length()) return;
+        }
```
Comment (marcj) [inline src/ResizeSensor.js]: Why is this necessary? The whole `element.resizedAttached` should be deleted, so I don't understand why you would want to remove only one event.

Comment (jsdw) [inline src/ResizeSensor.js]: This makes the following possible:

```
var fn1 = function(){ console.log("resize 1") }
ResizeSensor(element, fn1);

var fn2 = function(){ console.log("resize 2") }
ResizeSensor(element, fn2);

// at this point, resizing el would log
//
// resize 1
// resize 2
// resize 1
// resize 2
//...

ResizeSensor.detach(element, fn2);

// now, resizing would log only
//
// resize 1
// resize 1
// resize 1
//...
```

This sort of approach mirrors the native addEventListener/removeEventListener interface, which allows you to remove specific events by passing a ref to the original function.

In the event that you call `ResizeSensor.detach(element,fn)` and the fn removed leaves none left in the event queue, we do the rest of the clean up as well, else we leave the rest since it's necessary for the remaining events.

Comment (marcj) [inline src/ResizeSensor.js]: alright, thanks for explanation!

**Comment by marcj** [comment]
Thanks!
